### ISSUE TITLE
Render ldapURL value in attribute-resolver.xml and login.config

### ISSUE THREAD

**Issue by iromli** [GluuFederation/oxTrust#114]
In cluster edition, we have a usecase where we use 2 LDAP servers:

1. ldap-1: host `55422c825258` and port `1636`
2. ldap-2: host `cf96202f1860` and port `1636`

Given these 2 LDAP servers, cluster renders the following value in `oxTrust.properties`:

```
idp.ldap.server=55422c825258:1636,cf96202f1860:1636
```
which will generate `/opt/idp/conf/attribute-resolver.xml` as seen below:

```xml
    <!-- LDAP Connector -->
    <resolver:DataConnector id="siteLDAP" xsi:type="dc:LDAPDirectory"
        ldapURL="ldaps://55422c825258:1636,cf96202f1860:1636"
    </resolver:DataConnector>
```

Certainly, this is an invalid connection string since there's a comma character there.

I've tried to change `oxTrust.properties` to replace the comma with whitespace:

```
idp.ldap.server=55422c825258:1636 cf96202f1860:1636
```

This will generate `attribute-resolver.xml` like this:

```xml
   <!-- LDAP Connector -->
    <resolver:DataConnector id="siteLDAP" xsi:type="dc:LDAPDirectory"
        ldapURL="ldaps://55422c825258:1636 cf96202f1860:1636"
    </resolver:DataConnector>
```

From `/opt/idp/idp-process.log`, I can see that connection to LDAP `55422c825258:1636` (ldap-1) is worked. However, when I shutdown ldap-1, Shib will try to connect to `cf96202f1860:1636` (ldap-2), and the connection is failed due to incorrect connection string. 

After reading docs at https://wiki.shibboleth.net/confluence/display/SHIB2/ResolverLDAPDataConnector, I figured out when I change the connection string to:

```
ldapURL="ldaps://55422c825258:1636 ldaps://cf96202f1860:1636"
```

Shib able to connect to available ldap-2 even when ldap-1 is unavailable.

I may be wrong, but I guess we need to change the way oxTrust reads `idp.ldap.server` value in `oxTrust.properties`. For example, when `idp.ldap.server` uses comma:

```
idp.ldap.server = 55422c825258:1636,cf96202f1860:1636
```

or whitespace:

```
idp.ldap.server = 55422c825258:1636 cf96202f1860:1636
```
it should be recognized as 2 different LDAP servers, which should render:

```
ldapURL="ldaps://55422c825258:1636 ldaps://cf96202f1860:1636"
``` 

in `attribute-resolver.xml`.

Also I've found the similar issue in `/opt/idp/conf/login.config`.

**Comment by rahatjaan** [comment]
the issue is fixed and committed to master

**Comment by iromli** [comment]
@rahatjaan is it shipped in oxTrust v2.3.4 WAR file?

**Comment by rahatjaan** [comment]
it is in 2.4 release which is now deployed to ce-release

**Comment by iromli** [comment]
Is it possible to backport this fix to v2.3.x? Cluster still uses v2.3.x and we really need this fix because we have usecase with replicated LDAP.

/cc @nynymike

**Comment by yurem** [comment]
Are we speaking about this commit https://github.com/GluuFederation/oxTrust/commit/fc3413ae33f02938360b755eac6e94932d5db50d ?

I can merge it into 2.3.4 and rebuild oxTtust war if needed.

**Comment by iromli** [comment]
@yurem yes, I think so

**Comment by iromli** [comment]
@yurem any news about merging the fix to oxTrust v2.3.4?

**Comment by iromli** [comment]
Tested and worked in v2.3.4. 
Thanks.
